### DISCUSSION TITLE
Geckodriver dışında driver kullanmak

### DISCUSSION THREAD

**Discussion by Tetherium** [KebabLord/turkanime-indirici#69] [Genel] [answered]
Script için teşekkürler, yaklaşık 2 yıldır sorunsuzca kullanıyorum.
Pek çok sistemde firefox bulunmuyor, chrome'un yüklenme hızının daha yüksek olduğu da aşikar, script'de webdriver olarak mozilla dışında chromium, edge gibi birşey kullanmak daha mantıklı değil mi?

**Comment by KebabLord** [comment, marked as answer]
Aşağıdaki sebeplerden dolayı firefox dışında bir webdriver desteği getirmeyi düşünmüyorum:
- Firefox daha az ram tüketiyor
- Arayüzü daha kişiselleştirilebilir ki bu da potansiyel bir gui durumunda işe yarar
- Selenium'dan kurtulmaya çalışıyoruz, ona daha da bağımlı hale gelmeye değil

Lakin kurmasının daha kolay oluşu, sayfaları daha hızlı yüklemesi ve popülerliği gibi faktörlerden undetected_chromedriver gibi projelere göçmek de kulağa fena gelmiyor, kiosk özelliği gui için işe yarar olabilir. Yine de şimdilik planlamıyorum.

Script'i farklı driverlar ile kullanmak isteyenler scripti chrome veya bir başka webdriver kullanmak üzere kolaylıkla editleyebilirler

örneğin Anime(..., ...) objesi yaratılırken kendi oluşturduğunuz başka bir driver'ı paslayabilirsiniz:
```py
import undetected_chromedriver as uc
from turkanime_api import Anime

driver = uc.Chrome(headless=True)
driver.get("https://turkanime.co/kullanici/anonim")
animu = Anime(driver, "non-non-biyori")
```

**Reply by Tetherium** [reply]
çok teşekkürler
